### DISCUSSION TITLE
When using browser history, the x-model attribute doesn't keep the state in sync with the input element.

### DISCUSSION THREAD

**Discussion by emil-nasso** [alpinejs/alpine#3284] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2505

<div type='discussions-op-text'>

<sup>Originally posted by **emil-nasso** December 16, 2021</sup>
I have a page with a search-bar that is handled using alpine.js. If I write something in the input-field, navigate away from the page and then get back to the page using the "back"-feature in my browser, I will still see my old input in the input field by the components state doesn't reflect that.

Here is a simplified reproduction of this problem:
https://jsfiddle.net/ynju46dg/14/

It seems to be working this way in chrome but firefox and safari doesn't repopulate the form when I use the back-feature.

Is this possibly a bug in alpine, or is it chrome that does something weird?</div>

**Comment by 0xearl** [comment]
i think this is because the x-model state is empty at first therefore the output is empty, but when you start typing in the input again the output will sync

https://www.loom.com/share/cf723b360b3249df810e518d12c7476a

**Comment by tauzahmd** [comment]
I had also seen this behavior, I suspected that is how it works on chrome.

**Comment by markjaquith** [comment]
@0xearl is right.

Chrome is silently restoring these form field values, but without firing events on them. So Alpine has no idea.

In [this Chromium bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1057463), a Chrome dev says that it's intentional, and that they can't fire `onchange` events when filling these values, because that might have unintended side effects.

Not really sure Alpine core should be opening this can of worms.

But hey — the [Persist](https://alpinejs.dev/plugins/persist) plugin fixes it!

https://jsfiddle.net/784rsdq0/2/

**Comment by cachitas** [comment]
I am facing the same issue using `x-data="{ open: false }"` and `x-show` to open a dropdown. Clicking a link and then hitting the Back button returns me to the initial page with the dropdown open.

This happens using Chrome. Brave and Firefox behave as expected.

**Comment by SimoTod** [comment]
> I am facing the same issue using `x-data="{ open: false }"` and `x-show` to open a dropdown. Clicking a link and then hitting the Back button returns me to the initial page with the dropdown open.
> 
> This happens using Chrome. Brave and Firefox behave as expected.

@cachitas You should be able to use `@pageshow.window="open = false"`

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
